### ISSUE TITLE
Screens: appending suffix "_main_menu" at the end of `style_prefix` value causes weird text offset

### ISSUE THREAD

**Issue by hsandt** [renpy/renpy#5198]
**Context**

In order to place top main menu textbuttons in a different way as in the rest of the main menu navigation (e.g. on the main menu Load screen, or in game menu in general), I created a custom screen `navigation_main_menu` copied from `navigation`, with its own `style_prefix`: `navigation_main_menu`.

I observed that the text was strongly offset to the left. Trying other `style_prefix`es, I realized that only `_main_menu` caused the issue. Other names, including adding suffix `_invalid` to cause the style not to be defined, will not cause such an issue.

`style_suffix "main_menu"` doesn't cause the issue (it has other weird issues like `style_suffix "main_menu"` being OK in navigation but other values causing Exception: Style does not exist, and not being accepted under screen `use`, but that would be for another report).

**Repro**

```rpy
# screens.rpy
screen navigation():

    vbox:
        # try the following in order:
        style_prefix "navigation"
        style_prefix "navigation_invalid"
        style_prefix "navigation_main_menu"

    # ...

# ...

screen history():

    tag menu

    ## Avoid predicting this screen, as it can be very large.
    predict False

    use game_menu(_("History"), scroll=("vpgrid" if gui.history_height else "viewport"), yinitial=1.0):
        # try the following in order:
        style_prefix "history"
        style_prefix "history_invalid"
        style_prefix "history_main_menu"

    # ...
```

style_prefix "navigation"
![Navigation with style_prefix navigation](https://github.com/renpy/renpy/assets/1067772/24d1d2f9-1790-4231-ba7a-0561f962c487)

style_prefix "navigation_invalid"
![Navigation with style_prefix navigation_invalid](https://github.com/renpy/renpy/assets/1067772/0f811a85-95dd-4ddc-bddd-2fa6effed904)

style_prefix "navigation_main_menu"
![Navigation with style_prefix navigation_main_menu](https://github.com/renpy/renpy/assets/1067772/7cbc5fa4-9f19-4c65-aaa1-a7226f642bb2)

style_prefix "history"
![History with style_prefix history](https://github.com/renpy/renpy/assets/1067772/4ed4829c-2d13-4daf-b1b1-3c8f9b37dcba)

style_prefix "history_invalid"
![History with style_prefix history_invalid](https://github.com/renpy/renpy/assets/1067772/fe486176-872d-45b2-8fe7-a5eff4a8ea40)

style_prefix "history_main_menu"
![History with style_prefix history_main_menu](https://github.com/renpy/renpy/assets/1067772/ad265ef5-582b-4235-a384-ffc65369a88e)

This must not be a coincidence, so "main_menu" must be something special, but I don't see it documented anywhere.

**Workaround**

I will name my style `navigation_title_menu` instead.

**Comment by MiiNiPaa** [comment]
From documentation on [styles](https://renpy.org/doc/html/style.html)

> When a style is defined without a parent being specified, a default parent is chosen for the style. If the style contains an underscore (_) in its name, the parent is named by removing everything up to and including the first underscore. When a style that does not exist is used, and the style has an underscore in its name, Ren'Py will create it using the default parent. For example, a style named my_button will inherit from button. This inheritance can be changed using the style statement or using methods of the [Style](https://renpy.org/doc/html/style.html#Style) class. Otherwise, the style named "default" is used.

In your case it defines new styles inheriting from `main_menu_vbox`, `main_menu_button`, etc, some of which are autogenerated from next parent:
![](https://i.imgur.com/SnpJPax.png)

**Comment by hsandt** [comment]
Thanks, I also stumbled on `style main_menu_vbox:` as I was customising the main menu further. But hover UI element > Shift + I to inspect style is even faster!

So according to the rule, using two words before `_main_menu` should work. Strange, I thought I had tried `navigation_for_main_menu` at some point, and that it failed too...

So to sum up, I should just avoid reusing names used sic by other styles, but I can use anything else like `title_menu` or `top_menu`.

Since Renpy behaves as designed, I will close this comment. Hopefully the thread will still help other people struggling with this issue.
